### DISCUSSION TITLE
Prisma Storing Image over 1 mb giving Server has closed the connection

### DISCUSSION THREAD

**Discussion by formerskulk4** [prisma/prisma#21655] [Q&A]
### Bug description

I am trying to store image as blob using Prisma. On schema I have stored image as Bytes @db.MediumBlob, which has approximately 16MB. 
If I upload image less than 1 mb they are stored, if they are more than 1 mb is gives the error that
→ 321 await prisma.vehicle_Images.createMany(
**Server has closed the connection.**

### How to reproduce

Upload an image with more than 1mb of size, and I got this error that server has closed connection.

This is the code:

```
const UPLOAD_IMAGE = async (req, res) => {
  try {
    const form = formidable();

    form.on("error", (err) => {
      console.error("Error parsing form: ", err);
      return res.status(500).json({
        success: false,
        code: "A00010",
      });
    });

    form.parse(req, async (err, fields, files) => {
      try {

        const imagesToCreate = [];

        for (const file of filesToUpload) {
          console.log(file);
          const fileData = fs.readFileSync(file.filepath);

          imagesToCreate.push({
            image: fileData,
          });
        }

        await prisma.vehicle_Images.createMany({
          data: imagesToCreate,
        });

        return res.status(200).json({
          success: true,
        });
      } catch (error) {
        console.error("Error during image upload: ", error);
        return res.status(500).json({
          success: false,
          code: "A00010",
        });
      }
    });
  } catch (error) {
    console.error("Error in upload process: ", error);
    return res.status(500).json({
      success: false,
      code: "A00010",
    });
  }
};
```

### Expected behavior

If you upload an image more than 1mb it will say that server has closed connection, I tried with an image less than 1 mb it stores correctly.

### Prisma information

```prisma
model Vehicle_Images {
  id            String   @id @default(uuid())
  image         Bytes	   @db.LongBlob
}
```

### Environment & setup

- OS: Windows
- Database: mySQL
- Node.js version: v16.15.1

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.2.1
@prisma/client          : 5.5.2
Current platform        : windows
Query Engine (Node-API) : libquery-engine 2920a97877e12e055c1333079b8d19cee7f33826 (at ..\..\..\..\..\Users\Parid Varoshi\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 2920a97877e12e055c1333079b8d19cee7f33826 (at ..\..\..\..\..\Users\Parid Varoshi\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core 2920a97877e12e055c1333079b8d19cee7f33826 (at ..\..\..\..\..\Users\Parid Varoshi\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 2920a97877e12e055c1333079b8d19cee7f33826 (at ..\..\..\..\..\Users\Parid Varoshi\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 2920a97877e12e055c1333079b8d19cee7f33826
Studio                  : 0.469.0
Preview Features        : interactiveTransactions
```

**Comment by ludralph** [comment]
Hi @formerskulk4 👋 

Thank you for raising this question.

The error message `Server has closed the connection` typically indicates a problem with the database connection rather than an issue with Prisma itself.

MySQL has a system variable called `max_allowed_packet` that determines the maximum size of a packet or any generated/intermediate string. If the size of the image you're trying to upload exceeds this limit, it could cause the connection to close. You might need to increase the value of max_allowed_packet in your MySQL configuration. 
This setting controls the maximum size of a packet that can be sent to or received from the server. Increase it to accommodate larger blobs. For example, set it to 32M or higher:
```
max_allowed_packet = 32M
```
If you're using a server like Express.js, there might be a limit on the size of the request body. You might need to increase this limit to allow for larger image uploads.

Lastly, storing large files (like images) directly in a database is generally not recommended because it can quickly consume your database storage and impact performance. Instead, you could consider storing the images in a dedicated file storage service (like AWS S3, Google Cloud Storage, etc.) and then storing the URL or reference to the image in your database.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
